### DISCUSSION TITLE
eTag support

### DISCUSSION THREAD

**Discussion by danielebogo** [onevcat/Kingfisher#1632] [Q&A]
Hi there 👋 
I know from some [old closed issue](https://github.com/onevcat/Kingfisher/issues/35) there was a wiki page 👉 [here](https://github.com/onevcat/Kingfisher/wiki/How-to-implement-ETag-based-304-%28Not-Modified%29-handling-in-Kingfisher) to help implementing NSCache.
I need to replicate the `SDWebImageOptions.refreshCached` behaviour in some way.

Any help/idea?

**Comment by onevcat** [comment]
Hi,

I have to say the ETag support was never an "official" part of Kingfisher. It is using a URL-based way to cache the result.

I am not sure what `SDWebImageOptions.refreshCached` is doing. But it sounds like you need to force a refresh when loading the image. If that is true, [`.forcerefresh`](https://kingfisher.onevcat.com/kingfisheroptionsinfoitem/#kingfisheroptionsinfoitem.forcerefresh) might be identical, which will always load the image from the URL instead of using a cached one.

```swift
imageView.kf.setImage(with: url, options: [.forcerefresh])
```

**Reply by danielebogo** [reply]
Thanks @onevcat ! 
If I'm not wrong with that flag it uses NSCache instead of SDWebImage cache and deals with images changing behind the same request URL. But I have no idea if it displays the in memory image first while is fetching the HTTP response.

**Reply by danielebogo** [reply]
Btw I think I find a solution to what I'm trying to achieve. I always want the ImageFormat stored as image property and I think I can achieve that with a custom Processor.

**Comment by choli** [comment]
What I ended up, and seems to be a way to go, is to set two options:
`.forceRefresh,` // make sure to skip the Kingfisher cache
`.requestModifier(MyRequestModifier()),` // where we reset the cachePolicy on the URLRequest to the default `-useProtocolCachePolicy`.

Does that seem about right?
